chromium_src/chrome/installer/setup/install_worker_unittest.cc

```diff
@@ -0,0 +1,180 @@
+  MOCK_METHOD2(AddDeleteTreeWorkItem,
+               WorkItem*(const base::FilePath&, const base::FilePath&));
+  MOCK_METHOD4(AddMoveTreeWorkItem,
+               WorkItem*(const base::FilePath&,
+                         const base::FilePath&,
+                         const base::FilePath&,
+                         MoveTreeOption));
+  // Workaround for gmock problems with disambiguating between string pointers
+  // and DWORD.
+  WorkItem* AddSetRegValueWorkItem(HKEY a1,
+                                   const std::wstring& a2,
+                                   REGSAM a3,
+                                   const std::wstring& a4,
+                                   const std::wstring& a5,
+                                   bool a6) override {
+    return AddSetRegStringValueWorkItem(a1, a2, a3, a4, a5, a6);
+  }
+
+  MOCK_METHOD6(AddSetRegStringValueWorkItem,
+               WorkItem*(HKEY,
+                         const std::wstring&,
+                         REGSAM,
+                         const std::wstring&,
+                         const std::wstring&,
+                         bool));
+  MOCK_METHOD6(AddSetRegDwordValueWorkItem,
+               WorkItem*(HKEY,
+                         const std::wstring&,
+                         REGSAM,
+                         const std::wstring&,
+                         DWORD,
+                         bool));
+};
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+
+// Test for when registry key exists and value is already `1`.
+TEST_F(BraveSetupInstallWorkerTest, CleanupAlreadyRan) {
+  registry_util::RegistryOverrideManager override_manager;
+  std::wstring temp;
+  ASSERT_NO_FATAL_FAILURE(
+      override_manager.OverrideRegistry(HKEY_LOCAL_MACHINE, &temp));
+  ScopedNTRegistryTestingOverride nt_override(nt::HKLM, temp);
+
+  // Write out a value `1` (simulating already ran)
+  base::win::RegKey key(HKEY_LOCAL_MACHINE,
+                        brave_vpn::kBraveVpnOneTimeServiceCleanupStoragePath,
+                        KEY_ALL_ACCESS);
+  DWORD cleanup_ran = 1;
+  key.WriteValue(brave_vpn::kBraveVpnOneTimeServiceCleanupValue, cleanup_ran);
+
+  NiceMock<MockWorkItemList> work_item_list;
+  EXPECT_FALSE(installer::OneTimeServiceCleanup(example_path_, example_version_,
+                                                &work_item_list, true));
+}
+
+// Test for when no registry key exists yet.
+TEST_F(BraveSetupInstallWorkerTest, CleanupNotRanYetNoKey) {
+  registry_util::RegistryOverrideManager override_manager;
+  std::wstring temp;
+  ASSERT_NO_FATAL_FAILURE(
+      override_manager.OverrideRegistry(HKEY_LOCAL_MACHINE, &temp));
+  ScopedNTRegistryTestingOverride nt_override(nt::HKLM, temp);
+
+  NiceMock<MockWorkItemList> work_item_list;
+  EXPECT_TRUE(installer::OneTimeServiceCleanup(example_path_, example_version_,
+                                               &work_item_list, true));
+
+  // Ensure it set `ran` to `1`
+  base::win::RegKey key(HKEY_LOCAL_MACHINE,
```

Comment:
I would also extract the logic for reading `kBraveVpnOneTimeServiceCleanupStoragePath` into a function.

Reply:
same as above